### PR TITLE
adding some new float functions

### PR DESCRIPTION
**Pull request by dpzmick**
work in progress still, just want to open this to track progress and get any early feedback if anyone is feeling generous!

### CODE REVIEW

**Comment by DataAnalysisCosby** [comment]
One important first step should be replace the "docs go here" strings. 
Also hey David, funny to see you here. I also saw this and decided to contribute

**Comment by dpzmick** [comment]
Yes docs do need to go there!

What a coincidence lol. This is all your fault though. You told me rust was cool now I can't stop.

**Comment by DataAnalysisCosby** [comment]
Programming in rust is like programming for the first time again.

**Comment by Wilfred** [comment]
All looks sensible so far. Once we have functionality for making float values written in Rust we can write tests, but it's not really possible yet. 

Let me know when it's ready :)

**Comment by dpzmick** [comment]
@Wilfred how do you feel about this `lisp_fun` macro?

File: rust_src/src/lib.rs [hunk 1]
```diff
@@ -36,25 +36,25 @@ extern "C" {
...
-        defsubr(&*symbols::Ssymbolp);
-        defsubr(&*cons::Sconsp);
-        defsubr(&*cons::Ssetcar);
-        defsubr(&*cons::Ssetcdr);
-        defsubr(&*cons::Scar);
-        defsubr(&*strings::Sstringp);
-        defsubr(&*strings::Seq);
-        defsubr(&*strings::Snull);
-    }
+pub unsafe extern "C" fn rust_init_syms() {
```
Comment (Wilfred) [inline rust_src/src/lib.rs]: We really need some docs on our safety policy, I'll write something later. I think we should only have `unsafe` functions when it's possible to call them in ways that breaks memory safety. I don't think that's true here, so I propose we have an `unsafe` block inside the function.

File: rust_src/src/floatfns.rs [hunk 2]
```diff
@@ -26,3 +52,167 @@ pub fn fmod_float(x: LispObject, y: LispObject) -> LispObject {
...
+
+simple_float_op!("fceiling", Ffceiling, Sfceiling, ceil,
+    "Return the smallest integer no less than ARG, as a float.
+(Round toward +inf.)");
+
+simple_float_op!("ffloor", Ffloor, Sffloor, floor,
+    "Return the largest integer no greater than ARG, as a float.
+(Round towards -inf.)");
+
+lisp_fun! {
```
Comment (Wilfred) [inline rust_src/src/floatfns.rs]: We have a `defun!` macro now, could you use that instead?

I appreciate it didn't exist when you started on this PR, sorry about that.

Comment (dpzmick) [inline rust_src/src/floatfns.rs]: sure no problem

**Comment by dpzmick** [comment]
@Wilfred I think I've done all I'm going to do on these functions this time around. The rest will require a bit more work and this review is starting to get big. I can tackle the next set in a separate PR.

**Pull request merged**